dask/dataframe/io/parquet.py

```diff
@@ -174,11 +145,155 @@ def _read_parquet_row_group(open, fn, index, columns, rg, series, categories,
+
+    if not isinstance(all_columns, list):
+        out_type = Series
+        all_columns = [all_columns]
+    else:
+        out_type = DataFrame
+
+    divisions = (None,) * (len(dataset.pieces) + 1)
+
+    dtypes = _get_pyarrow_dtypes(schema)
+
+    meta = _meta_from_dtypes(all_columns, schema.names, dtypes)
+
+    task_plan = {
+        (task_name, i): (_read_arrow_parquet_piece,
+                         file_opener,
+                         piece, all_columns,
+                         out_type == Series,
+                         dataset.partitions)
+        for i, piece in enumerate(dataset.pieces)
+    }
+
+    return out_type(task_plan, task_name, meta, divisions)
+
+
+def _get_pyarrow_dtypes(schema):
+    dtypes = {}
+    for i in range(len(schema)):
+        field = schema[i]
+        numpy_dtype = field.type.to_pandas_dtype()
+        dtypes[field.name] = numpy_dtype
+
+    return dtypes
+
+
+def _read_arrow_parquet_piece(open_file_func, piece, columns, is_series,
+                              partitions):
+    with open_file_func(piece.path, mode='rb') as f:
+        table = piece.read(columns=columns,  partitions=partitions,
+                           file=f)
+    df = table.to_pandas()
+
+    if is_series:
+        return df[df.columns[0]]
+    else:
+        return df
+
+
+# ----------------------------------------------------------------------
+# User read API
+
+def read_parquet(path, columns=None, filters=None, categories=None, index=None,
+                 storage_options=None, engine='auto'):
+    """
+    Read ParquetFile into a Dask DataFrame
+
+    This reads a directory of Parquet data into a Dask.dataframe, one file per
+    partition.  It selects the index among the sorted columns if any exist.
+
+    Parameters
+    ----------
+    path : string
+        Source directory for data. May be a glob string.
+        Prepend with protocol like ``s3://`` or ``hdfs://`` for remote data.
+    columns: list or None
+        List of column names to load
+    filters: list
+        List of filters to apply, like ``[('x', '>' 0), ...]``
+    index: string or None (default) or False
+        Name of index column to use if that column is sorted;
+        False to force dask to not use any column as the index
+    categories: list, dict or None
+        For any fields listed here, if the parquet encoding is Dictionary,
+        the column will be created with dtype category. Use only if it is
+        guaranteed that the column is encoded as dictionary in all row-groups.
+        If a list, assumes up to 2**16-1 labels; if a dict, specify the number
+        of labels expected; if None, will load categories automatically for
+        data written by dask/fastparquet, not otherwise.
+    storage_options : dict
+        Key/value pairs to be passed on to the file-system backend, if any.
+    engine : {'auto', 'fastparquet', 'arrow'}, default 'auto'
+        Parquet reader library to use. If only one library is installed, it
```

Comment:
This phrasing seems wrong. Presumably if I say "arrow" but it isn't around, there is an error, and the If Only One relates to the 'auto' choice.